File: core/src/main/java/mifos/org/mobilewallet/core/domain/usecase/VerifyUser.java
package mifos.org.mobilewallet.core.domain.usecase;

import javax.inject.Inject;

import mifos.org.mobilewallet.core.base.UseCase;
import mifos.org.mobilewallet.core.data.fineract.entity.register.UserVerify;
import mifos.org.mobilewallet.core.data.fineract.repository.FineractRepository;
import okhttp3.ResponseBody;
import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by naman on 17/8/17.
 */

public class VerifyUser extends UseCase<VerifyUser.RequestValues,
        VerifyUser.ResponseValue> {

    private final FineractRepository apiRepository;

    @Inject
    public VerifyUser(FineractRepository apiRepository) {
        this.apiRepository = apiRepository;
    }


    @Override
    protected void executeUseCase(VerifyUser.RequestValues requestValues) {

        apiRepository.verifyUser(requestValues.userVerify)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe(new Subscriber<ResponseBody>() {
                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {
                        getUseCaseCallback().onError("Error verifying user");
                    }

                    @Override
                    public void onNext(ResponseBody responseBody) {
                        getUseCaseCallback().onSuccess(new ResponseValue());
                    }
                });

    }


    public static final class RequestValues implements UseCase.RequestValues {

        private final UserVerify userVerify;

        public RequestValues(UserVerify userVerify) {
            this.userVerify = userVerify;
        }
    }

    public static final class ResponseValue implements UseCase.ResponseValue {

        public ResponseValue() {

        }

    }
}

File: core/data/src/main/java/org/mifospay/core/data/domain/usecase/user/VerifyUser.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
 */
package org.mifospay.core.data.domain.usecase.user

import com.mifospay.core.model.entity.register.UserVerify
import okhttp3.ResponseBody
import org.mifospay.core.data.base.UseCase
import org.mifospay.core.data.fineract.repository.FineractRepository
import org.mifospay.core.data.util.Constants
import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class VerifyUser (
    private val apiRepository: FineractRepository,
) : UseCase<VerifyUser.RequestValues, VerifyUser.ResponseValue>() {
    override fun executeUseCase(requestValues: RequestValues) {
        apiRepository.verifyUser(requestValues.userVerify)
            .observeOn(AndroidSchedulers.mainThread())
            .subscribeOn(Schedulers.io())
            .subscribe(
                object : Subscriber<ResponseBody?>() {
                    override fun onCompleted() {}
                    override fun onError(e: Throwable) {
                        useCaseCallback.onError(Constants.ERROR_VERIFYING_USER)
                    }

                    override fun onNext(t: ResponseBody?) {
                        useCaseCallback.onSuccess(ResponseValue())
                    }
                },
            )
    }

    class RequestValues(val userVerify: UserVerify) : UseCase.RequestValues
    class ResponseValue : UseCase.ResponseValue
}
